### PR TITLE
chore: (kernel) submission cascade delete

### PR DESCRIPTION
**Pull request by lordmallam**
Remove cascade delete on mappingset ==> submissions

### CODE REVIEW

File: aether-kernel/aether/kernel/api/serializers.py [hunk 1]
```diff
@@ -203,7 +203,7 @@ class AttachmentSerializer(DynamicFieldsMixin, serializers.ModelSerializer):
 
     submission = MtPrimaryKeyRelatedField(
         queryset=models.Submission.objects.all(),
-        mt_field='mappingset__project',
+        mt_field='project',
```
Comment (obdulia-losantos) [inline aether-kernel/aether/kernel/api/serializers.py:210]: Also in the `class SubmissionSerializer` the `mappingset` field is no longer required ;-)

Comment (obdulia-losantos) [reply aether-kernel/aether/kernel/api/serializers.py:210]: But I would say that **IT IS REQUIRED TO CREATE THE SUBMISSION** or no entities will be extracted...

**Comment by obdulia-losantos** [comment]
Another point, what happens with the entity extraction if the submission has no mappingset? :scream:

**Comment by lordmallam** [comment]
@obdulia-losantos a mappingset is required on submission create only

**Comment by obdulia-losantos** [comment]
Have you tried to update a submission without mapping set?

File: aether-kernel/aether/kernel/api/serializers.py [hunk 2]
```diff
@@ -168,13 +168,11 @@ class SubmissionSerializer(DynamicFieldsMixin, serializers.ModelSerializer):
         queryset=models.Project.objects.all(),
         required=False,
     )
-    mappingset = MtPrimaryKeyRelatedField(
```
Comment (obdulia-losantos) [inline aether-kernel/aether/kernel/api/serializers.py]: You cannot remove this or multitenancy will not work properly :disappointed: you only need to add the option `required=False`

File: aether-kernel/aether/kernel/api/serializers.py [hunk 3]
```diff
@@ -168,13 +168,11 @@ class SubmissionSerializer(DynamicFieldsMixin, serializers.ModelSerializer):
...
-        queryset=models.MappingSet.objects.all(),
-        mt_field='project',
-    )
 
     def create(self, validated_data):
         instance = super(SubmissionSerializer, self).create(validated_data)
+        if instance.mappingset is None:
```
Comment (obdulia-losantos) [inline aether-kernel/aether/kernel/api/serializers.py]: The other way around, check the value before create it
```python
        if validated_data.get('mappingset') is None:
            raise serializers.ValidationError('Mappingset must be provided on initial submission')
        instance = super(SubmissionSerializer, self).create(validated_data)
```

**Review by obdulia-losantos — APPROVED** [review]
:see_no_evil:

**Pull request merged**